Sources/MapboxCoreNavigation/LegacyRouteController.swift

```diff
@@ -58,12 +58,12 @@ open class LegacyRouteController: NSObject, Router, InternalRouter, CLLocationMa
     }
     
     public var indexedRoute: IndexedRoute {
-        get {
-            return routeProgress.indexedRoute
-        }
-        set {
-            routeProgress.indexedRoute = newValue
-        }
+        routeProgress.indexedRoute
+    }
+
+    public func updateRoute(with indexedRoute: IndexedRoute, routeOptions: RouteOptions?) {
+        let routeOptions = routeOptions ?? routeProgress.routeOptions
+        routeProgress = RouteProgress(route: indexedRoute.0, routeIndex: indexedRoute.1, options: routeOptions)
```

Comment:
shouldn't we also `announce(reroute:)` here as it was previously?

Reply:
An announce happens in `routeProgress` setter.